lte/gateway/c/session_manager/Utilities.cpp

```diff
@@ -63,4 +63,11 @@ std::chrono::milliseconds time_difference_from_now(
   return std::chrono::duration_cast<std::chrono::milliseconds>(sec);
 }
 
+std::string prepend_imsi_with_prefix(const std::string& imsi) {
+  if (imsi.find(IMSI_PREFIX) == std::string::npos) {
+    return IMSI_PREFIX + imsi;
+  }
+  return imsi;
+}
+
```

Comment:
We already went through this prepend thing in another PR 
https://github.com/magma/magma/pull/10825#issuecomment-997715173

As shown in another PR, sessiond must receive IMSIS properly formatted, same way we do at 4g.

Reply:
Yeah, Agreed with your comment, all the communication between AMF - SMF will be handled by AMF code with IMSI prefix. UPF deosn't have IMSI information, it will have only UE IP address. This particular code is when UPF receives paging request it fetches imsi information from Mobilityd with UE IP address with use of below method
`mobilityd_client_->get_subscriberid_from_ipv4()` in `UpfMsgManageHandler.cpp`. Here Mobilityd is sending imsi value without IMSI prefix, so problem came here.

So here two possibilities like bleow:
1) add IMSI prefix in sessiond as proposed in this PR
2) Mobilityd to send imsi with IMSI prefix

So kindly let us know which one you suggest among above two possibilities.

Reply:
Then I think it `get_subscriberid_from_ipv4` should be the one  where data is massaged so sessiond can ingest it.  

Since sessiond uses `IMSI12345` format `get_subscriberid_from_ipv4` will work in the future on other pleaces it is used in sessiond.

Reply:
note this function must also check if prefix was not already added, just in case

Reply:
`get_subscriberid_from_ipv4` this function used in 4G and 5G in many places so we cannot the return string from this function. So we modified and introduced this sessiond code which is checking IMSI prefix if is not there we are adding IMSI prefix.